src/core/server/saved_objects/service/lib/integration_tests/repository_with_proxy_utils.ts

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import Hapi from '@hapi/hapi';
+import { IncomingMessage } from 'http';
+import { kibanaPackageJson as pkg } from '@kbn/utils';
+
+// proxy setup
+const defaultProxyOptions = (hostname: string, port: number | string) => ({
+  host: hostname,
+  port,
+  protocol: 'http' as 'http',
+  passThrough: true,
+});
+
+let proxyInterrupt: string | null | undefined = null;
```

Comment:
Perhaps as a middle ground for now, we could at least type `setProxyInterrupt` to take a fixed list of string literals instead of just `string`? e.g.
```ts
export const setProxyInterrupt = (testArg: 'foo' | 'bar' | 'baz') => {...}
```

Also, is `null | undefined` necessary here, or could it be one or the other?